bindings/nodejs/index.d.ts

```diff
@@ -64,6 +64,24 @@ export class Operator {
    * An error will be returned if given path doesn't end with `/`.
    */
   listSync(path: string): BlockingLister
+  /**
+   * Get a presigned request for read.
+   *
+   * Unit of expires is seconds.
+   */
+  presignRead(path: string, expires: number): PresignedRequest
+  /**
+   * Get a presigned request for write.
+   *
+   * Unit of expires is seconds.
+   */
+  presignWrite(path: string, expires: number): PresignedRequest
+  /**
+   * Get a presigned request for stat.
+   *
+   * Unit of expires is seconds.
+   */
+  presignStat(path: string, expires: number): PresignedRequest
```

Comment:
I'm hesitant about the API. Could you please write a simple demo to show how a JavaScript or TypeScript developer can use this API? The most important thing for me is whether they will enjoy using `opendal`.

Reply:
Example added. 

/cc @promer94

Reply:
## Benefit
The benefit of presigned url is that data could be safely downloaded (uploaded) from(to) data storage without passing through the server.  
It can save a lot of traffic and reduce the server load.

## Example usage

* Redirect the client to download the data directly from the data storage
```js
const server = http.createServer(async (req, res) => {
  if (req.url.startsWith('/download') && req.method === 'GET') {
    const urlParts = url.parse(req.url, true)
    const path = urlParts.query.path
    const expires = urlParts.query.expires

    const presignedRequest = op.presignRead(path, parseInt(expires))

    res.statusCode = 307
    res.setHeader('Location', presignedRequest.uri)
    res.end(end)
  } else {
    res.statusCode = 404
    res.end('Not Found')
  }
})
```


* Pass the `presignedRequest` to the client, client could build a stander web `request` from it.
```js
const request = new Request(presignedRequest.uri, {
   headers: presignedRequest.headers,
   method: presignedRequest.method
})
fetch(request)
```

Reply:
> ```js
> const request = new Request(presignedRequest.uri, {
>    headers: presignedRequest.headers,
>    method: presignedRequest.method
> })
> fetch(request)
> ```

Cool, this is what I want. So users can pass `presignedRequest.headers` to `Request`'s construct without anyother changes?

Reply:
> Cool, this is what I want. So users can pass presignedRequest.headers to Request's construct without changes?

Yes, the allowed header format is 
```ts
type HeadersInit = [string, string][] | Record<string, string> | Headers;
```
<img width="463" alt="CleanShot 2023-03-26 at 19 57 16@2x" src="https://user-images.githubusercontent.com/15878786/227774032-c13a9cb9-b1cc-4cbd-9ee7-bd0c4d7e46c7.png">

Reply:
Cool!